pytest/lib/mocknet.py

```diff
@@ -167,43 +212,60 @@ def chain_measure_bps_and_tps(archival_node,
         end_time = curr_time
     if start_time is None:
         start_time = end_time - duration
+    logger.info(
+        f'Measuring BPS and TPS in the time range {start_time} to {end_time}')
 
-    # one entry per block, equal to the timestamp of that block
+    # One entry per block, equal to the timestamp of that block.
     block_times = []
-    # one entry per block (because there is only one shard), equal to the number of transactions
+    # One entry per block, containing the count of transactions in all chunks of the block.
     tx_count = []
+    block_counter = 0
     while curr_time > start_time:
         if curr_time < end_time:
             block_times.append(curr_time)
-            chunk_hash = curr_block['chunks'][0]['chunk_hash']
-            chunk = archival_node.get_chunk(chunk_hash)['result']
-            tx_count.append(len(chunk['transactions']))
+            txs = 0
+            tx_per_chunk = [None] * len(curr_block['chunks'])
+            pmap(
+                lambda i: get_chunk_txn(i, curr_block['chunks'], archival_node,
+                                        tx_per_chunk),
+                range(len(curr_block['chunks'])))
+            for stat in tx_per_chunk:
+                txs += stat
+            tx_count.append(txs)
+            logger.info(
+                f'Processed block at time {curr_time} height #{curr_block["header"]["height"]}, # txs in a block: {txs}, per chunk: {tx_per_chunk}'
+            )
         prev_hash = curr_block['header']['prev_hash']
         curr_block = archival_node.get_block(prev_hash)['result']
         curr_time = get_timestamp(curr_block)
     block_times.reverse()
     tx_count.reverse()
+    assert block_times
     tx_cumulative = data.compute_cumulative(tx_count)
     bps = data.compute_rate(block_times)
     tps_fit = data.linear_regression(block_times, tx_cumulative)
+    logger.info(
+        f'Num blocks: {len(block_times)}, num transactions: {len(tx_count)}, bps: {bps}, tps_fit: {tps_fit}'
+    )
     return {'bps': bps, 'tps': tps_fit['slope']}
 
 
-def get_tx_events_single_node(node):
+def get_tx_events_single_node(node, tx_filename):
     try:
-        m = node.machine
-        target_file = f'./logs/{m.name}_txs'
-        m.download(TX_OUT_FILE, target_file)
-        with open(target_file) as f:
+        target_file = f'./logs/{node.instance_name}_txs'
+        node.machine.download(tx_filename, target_file)
+        with open(target_file, 'r') as f:
             return [float(line.strip()) for line in f.readlines()]
-    except:
+    except Exception as e:
+        logger.info(f'Getting tx_events from {node.instance_name} failed: {e}')
```

Comment:
Perhaps `logger.exception` instead? You won’t need to pass the exception explicitly then.